numpy/core/src/common/array_assign.c

```diff
@@ -125,9 +125,13 @@ raw_array_is_aligned(int ndim, npy_intp *shape,
 
         return npy_is_aligned((void *)align_check, alignment);
     }
-    else {
+    else if (alignment == 1) {
         return 1;
     }
+    else {
+        /* always return false for alignment == 0, which means unaligned */
+        return 0;
```

Comment:
Why is `alignment == 0` even a legal argument to this function?

Reply:
Maybe there is a better way to organize it... but the idea is that `npy_uint_alignment`, which normally returns the uint alignment, returns 0 if the type is not uint-alignable.

Then, it turned out to be convenient to allow the result of `npy_uint_alignment` to be passed directly to `raw_array_is_aligned`. 

An alternative would be to check the return value of `npy_uint_alignment` explicitly everywhere it is used. But since it is pretty much always used as an arg to `raw_array_is_aligned` it seemed less complicated this way.

Reply:
changed the comment